docs/guide/javascript_and_css.md

```diff
@@ -8,6 +8,199 @@ parent: How-to guide
+├── ...
+├── examples
+|   ├── hello_world_component
+|   |   ├── hello_world_component_controller.js
+|   |   ├── hello_world_component.css
+|   |   └── hello_world_component.html.erb
+|   └── hello_world_component.rb
+├── ...
+```
+
+#### 1. Prepare _app/components_ as an asset path for css and ensure hot reloads of Stimulus JavaScript
+
+```ruby
+# config/application.rb
+config.assets.paths << "app/components"
+config.importmap.cache_sweepers << config.root.join("app/components")
+```
+
+#### 2. Pin ViewComponent Stimulus import map entries
+
+```ruby
+# config/importmap.rb
+pin_all_from "app/components"
+```
+
+#### 3. Expose the Stimulus controller with a named key:
+
+```ruby
+# app/javascript/controllers/index.js
+import HelloWorldComponentController from "examples/hello_world_component/hello_world_component_controller"
+application.register("examples--hello-world-component", HelloWorldComponentController)
+```
+
+#### 4. Implement the ViewComponent with custom CSS and Stimulus behaviour:
+
+```ruby
+# app/components/examples/hello_world_component.rb
+class Examples::HelloWorldComponent < ViewComponent::Base
+  def initialize(title:)
+    @title = title
+    super
+  end
+end
+```
+
+```erb
+<!-- app/components/examples/hello_world_component/hello_world_component.html.erb -->
+<%= stylesheet_link_tag "examples/hello_world_component/hello_world_component" %>
+
+<h1><%= @title %></h1>
+<p><%= content %></p>
+
+<div data-controller="examples--hello-world-component">
+  <p class="hello-world" id="<%= @id %>" data-examples--hello-world-component-target="output">
+    This div will be updated by the controller
+  </p>
+
+  <button data-action="click->examples--hello-world-component#greet">Toggle Greeting</button>
+</div>
+```
+
+```css
+/* app/components/examples/hello_world_component/hello_world_component.css */
+.hello-world {
+  color: blue;
+}
+```
+
+```js
+// app/components/examples/hello_world_component/hello_world_component_controller.js
+import { Controller } from "@hotwired/stimulus";
+
+export default class extends Controller {
+  static targets = ["output"]
+
+  initialize() {
+    console.log("Component initialized!");
+  }
+
+  connect() {
+    console.log("Component connected!");
+
+    this.outputTarget.textContent = "This div has been initialised by stimulus and will be updated when you click the button"
+  }
+
+  greet() {
+    const currentText = this.outputTarget.textContent;
+    this.outputTarget.textContent = currentText === "Hello from Stimulus!"
+      ? "Goodbye from Stimulus!"
+      : "Hello from Stimulus!";
+  }
+}
+```
+
+#### 5. Render the component in a rails view (or a [ViewComponent preview](previews.md)) to see the end result:
+
+```erb
+<!-- app/views/layouts/application.html.erb -->
+<body>
+  ...
+    <%= render(Examples::HelloWorldComponent.new(title: "Hello World!")) {
+      "<em>This</em> will demonstrate the use of <b>Stimulus</b> and <b>CSS</b> in a ViewComponent".html_safe
+      }
+    %>
```

Comment:
No need to use `.html_safe` 😄 

```suggestion
    <%= render(Examples::HelloWorldComponent.new(title: "Hello World!")) do %>
      <em>This</em> will demonstrate the use of <b>Stimulus</b> and <b>CSS</b> in a ViewComponent
    <% end %>
```

Reply:
Without `.html_safe`, it renders like this:
<img width="572" alt="Screenshot 2024-12-20 at 7 42 50 am" src="https://github.com/user-attachments/assets/50595d8d-6f9d-49f2-b9e6-b4214b9508cf" />

Reply:
❤️ it!